sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py

```diff
@@ -158,6 +177,41 @@ def __init__(
         self._auth_uri = None
         self._properties = create_properties(self._config.user_agent)
 
+    @classmethod
+    def _convert_connection_string_to_kwargs(cls, conn_str, **kwargs):
+        # type: (str, Any) -> Dict[str, Any]
+        host, policy, key, entity_in_conn_str, token, token_expiry = _parse_conn_str(conn_str)
+        queue_name = kwargs.get("queue_name")
+        topic_name = kwargs.get("topic_name")
+        if not (queue_name or topic_name or entity_in_conn_str):
+            raise ValueError("Entity name is missing. Please specify `queue_name` or `topic_name`"
+                             " or use a connection string including the entity information.")
+
+        if queue_name and topic_name:
+            raise ValueError("`queue_name` and `topic_name` can not be specified simultaneously.")
+
+        entity_in_kwargs = queue_name or topic_name
+        if entity_in_conn_str and entity_in_kwargs and (entity_in_conn_str != entity_in_kwargs):
+            raise ServiceBusAuthenticationError(
+                "Entity names do not match, the entity name in connection string is {};"
+                " the entity name in parameter is {}.".format(entity_in_conn_str, entity_in_kwargs)
+            )
+
+        kwargs["fully_qualified_namespace"] = host
+        kwargs["entity_name"] = entity_in_conn_str or entity_in_kwargs
+        # This has to be defined seperately to support sync vs async credentials.
+        kwargs["credential"] = cls._create_credential_from_connection_string_parameters(token,
+                                                                                         token_expiry,
+                                                                                         policy,
+                                                                                         key)
+        return kwargs
+
+    @classmethod
+    def _create_credential_from_connection_string_parameters(cls, token, token_expiry, policy, key):
+        if token and token_expiry:
+            return ServiceBusSASTokenCredential(token, token_expiry)
+        return ServiceBusSharedKeyCredential(policy, key)
```

Comment:
not blocking this PR: wondering if it would be better to put this method into utils.

Reply:
as elsewhere https://github.com/Azure/azure-sdk-for-python/issues/13683